R/derive_param_tte.R

```diff
@@ -613,17 +622,31 @@ filter_date_sources <- function(sources,
       var = !!source_date_var,
       dataset_name = sources[[i]]$dataset_name
     )
-    data[[i]] <- source_dataset %>%
-      filter_if(sources[[i]]$filter) %>%
-      filter_extreme(
-        order = exprs(!!source_date_var),
-        by_vars = expr_c(subject_keys, by_vars),
-        mode = mode,
-        check_type = "none"
-      )
-
+    # wrap filter_extreme in tryCatch to catch duplicate records and create a message
+    data[[i]] <- tryCatch(
+      {
+        source_dataset %>%
+          filter_if(sources[[i]]$filter) %>%
+          arrange(!!!sources[[i]]$order) %>% # Ensure order is applied
+          filter_extreme(
+            order = exprs(!!source_date_var),
+            by_vars = expr_c(subject_keys, by_vars),
+            mode = mode,
+            check_type = check_type
+          )
+      },
+      warning = function(wrn) {
```

Comment:
This catches warnings only. Errors and messages should be handled as well. (Maybe https://rlang.r-lib.org/reference/try_fetch.html is helpful.)